models/agent_model.go

```diff
@@ -170,6 +171,9 @@ func (s *Agent) BeforeInsert() error {
 	if len(s.CustomLabels) == 0 {
 		s.CustomLabels = nil
 	}
+	if s.Status == "" && s.AgentType != ExternalExporterType && s.PMMAgentID != nil {
```

Comment:
```suggestion
	if s.Status == "" && s.AgentType != PMMAgentType {
```

Reply:
We need avoid both cases, External exporter and PMM Agent.